client/js/lounge.js

```diff
@@ -646,7 +672,15 @@ $(function() {
 		}
 	});
 
-	socket.on("names", renderChannelUsers);
+	socket.on("names", function(data) {
+		renderChannelUsers(data);
+
+		// Condensed actions support
+		var messages = chat.find("#chan-" + data.id).find(".messages");
+		data.users.forEach(function(user) {
+			messages.find(".user[data-name='" + user.name + "']").removeClass("disconnected");
```

Comment:
Ah, wait. Since condensed message is regenerating each time some action happens, we need to update users after that. Also, we can't take data from userlist, because it's updated later by this event.

But I think `setTimeout` is still needed here.